### DISCUSSION TITLE
How to add Apollo @cacheControl directive

### DISCUSSION THREAD

**Discussion by gapetto** [MichalLytek/typegraphql-nestjs#26] [General]
I'm trying to implement server side caching, following this documentation: https://www.apollographql.com/docs/apollo-server/performance/caching/.  Is there an example on how to add custom directives, so that the appear in generated schema?

**Comment by MichalLytek** [comment]
https://typegraphql.com/docs/directives.html

**Reply by gapetto** [reply]
Still can't figure out how to create directive the way the docs from that link show and then use it as an input in TypeGraphQLModule.forRoot(). I'm trying to build the directive using the class GraphQLDirective from grapql, but  with no success so far. Don't really know what locations mean and still trying to figure out how to assign to args...

**Reply by MichalLytek** [reply]
You only put `@Directive()` decorator over your field and queries. The directives will be living inside the `GraphQLSchema` it builds under the hood, so that Apollo Server sees it. 

As stated, they do not appear in the printed schema file.

Here you can find a middleware based approach, which at least work with plain typegraphql and apollo server:
https://github.com/MichalLytek/type-graphql/blob/v1.1.1/examples/apollo-engine/cache-control.ts
